src/orion/benchmark/task/profet/profet_task.py

```diff
@@ -0,0 +1,817 @@
+""" Base class for Tasks that are generated using the Profet algorithm.
+
+```
+@misc{klein2019metasurrogate,
+      title={Meta-Surrogate Benchmarking for Hyperparameter Optimization}, 
+      author={Aaron Klein and Zhenwen Dai and Frank Hutter and Neil Lawrence and Javier Gonzalez},
+      year={2019},
+      eprint={1905.12982},
+      archivePrefix={arXiv},
+      primaryClass={cs.LG}
+}
+```
+"""
+import functools
+import hashlib
+import json
+import os
+import pickle
+import warnings
+from contextlib import contextmanager
+from copy import deepcopy
+from dataclasses import asdict, dataclass, is_dataclass
+from logging import getLogger as get_logger
+from pathlib import Path
+from typing import Callable, Dict, Generic, List, Optional, Tuple, TypeVar, Union
+
+import GPy
+import numpy as np
+import torch
+from emukit.examples.profet.meta_benchmarks.architecture import get_default_architecture
+from emukit.examples.profet.meta_benchmarks.meta_forrester import (
+    get_architecture_forrester,
+)
+from emukit.examples.profet.train_meta_model import download_data
+from GPy.models import BayesianGPLVM
+from numpy.random import RandomState
+from pybnn.bohamiann import Bohamiann
+from torch import nn
+from torch.distributions import Normal
+from torch.functional import Tensor
+
+from orion.algo.space import Space
+from orion.benchmark.task.base import BaseTask
+from orion.core.io.space_builder import SpaceBuilder
+from orion.core.utils.format_trials import dict_to_trial, trial_to_tuple
+from orion.core.utils.points import flatten_dims
+from orion.core.worker.trial import Trial
+
+logger = get_logger(__name__)
+
+
+get_architecture: Dict[str, Callable[[int], nn.Module]] = dict(
```

Comment:
Now that each task has its own class definition, do you think it would make sense to move these configurations to each class and turn the functions below into methods of the base profet task class? I think this would organize better the information, instead of having part of it in each class (search space) and the rest in the main module.

Reply:
Hmm I would agree, but this would have to wait until another PR

Reply:
Maybe, but I'm afraid that is the kind of improvement we would never have to motivation to come back to if we delay it. :thinking:

Reply:
Ok, finally done with this, took quite a bit of work to do. Let me know what you think.

Reply:
It's great! Thanks a lot for taking the time to make these changes!